plugins/package-managers/bazel/src/main/kotlin/Bazel.kt

```diff
@@ -0,0 +1,309 @@
+                        type = managerName,
+                        name = moduleMetadata.module?.name ?: "root",
+                        version = moduleMetadata.module?.version.orEmpty(),
+                        namespace = ""
+                    ),
+                    definitionFilePath = VersionControlSystem.getPathInfo(definitionFile).path,
+                    authors = emptySet(),
+                    declaredLicenses = emptySet(),
+                    vcs = projectVcs,
+                    vcsProcessed = projectVcs,
+                    homepageUrl = "",
+                    scopeDependencies = scopes
+                ),
+                packages = packages.toSet()
+            )
+        )
+    }
+
+    private fun getPackages(
+        graph: Set<PackageReference>,
+        bmrClient: BazelModuleRegistryClient,
+        seen: MutableSet<Identifier> = mutableSetOf()
+    ): Set<Package?> {
+        val result = mutableSetOf<Package?>()
+
+        graph.forEach { ref ->
+            if (ref.id !in seen) {
+                seen.add(ref.id)
+
+                if (ref.id.name != "<root>") {
+                    result.add(packageFromPackageReference(ref, bmrClient))
+                }
+                result.addAll(getPackages(ref.dependencies, bmrClient, seen))
+            }
+        }
+
+        return result
+    }
+
+    private fun packageFromPackageReference(ref: PackageReference, bmrClient: BazelModuleRegistryClient): Package? {
+        val metadata: ModuleMetadata
+        try {
+            metadata = runBlocking { bmrClient.getModuleMetadata(ref.id.name) }
+        } catch (e: HttpException) {
+            logger.warn { "Failed to fetch metadata for Bazel module ${ref.id.name}: $e" }
+            return null
+        }
+
+        val sourceInfo: ModuleSourceInfo
+        try {
+            sourceInfo = runBlocking { bmrClient.getModuleSourceInfo(ref.id.name, ref.id.version) }
+        } catch (e: HttpException) {
+            logger.warn { "Failed to fetch source info for Bazel module ${ref.id.name}@${ref.id.version}: $e" }
+            return null
+        }
+
+        return Package(
+            id = ref.id,
+            declaredLicenses = emptySet(),
+            description = "",
+            homepageUrl = metadata.homepage?.toString().orEmpty(),
+            binaryArtifact = RemoteArtifact.EMPTY,
+            sourceArtifact = sourceInfo.remoteArtifact(),
+            vcs = toVcsInfo(metadata)
+        )
+    }
+
+    private fun getDependencyGraph(
+        projectDir: File,
+        depDirectives: Map<String, BazelDepDirective>,
+        lockFile: LockFile?
+    ): Set<Scope> {
+        if (lockFile != null) {
+            return parseLockFileGraph(lockFile.moduleDepGraph, depDirectives, managerName)
+        } else {
+            val modGraphProcess = run("mod", "graph", "--output", "json", workingDir = projectDir)
+            return parseBzlmodGraph(modGraphProcess.stdout, depDirectives, managerName)
```

Comment:
Can you help me understanding why you decided to use two code paths for construction the graph?

In particular, why isn't `mod graph` always used?

Reply:
It is not needed when a lock file is present, since that contains a dependency graph under `moduleDepGraph`. The usual approach in ORT as far as I can tell is to use the lock file when it is available.

Reply:
Hm, I'm not convinced from that yet. Do both approaches (lockfile parsing vs. module graph) yield the exact same information?

Reply:
They yield the same result for the dependencies that are declared with `bazel_dep` in `MODULE.bazel`, but not for all dependencies. The lock file contains more "Bazel internal" extra stuff that possibly nobody who uses ORT for Bazel will care about.

I'm happy to get rid of the code that extracts the dependency tree from the lock file.

Reply:
I'd appreciate if the lockfile code was removed then.